src/Umbraco.Web/HtmlHelperRenderExtensions.cs

```diff
@@ -355,8 +352,10 @@ public static MvcForm BeginUmbracoForm(this HtmlHelper html, string action, stri
                                                IDictionary<string, object> htmlAttributes,
                                                FormMethod method)
         {
-            if (string.IsNullOrWhiteSpace(action)) throw new ArgumentNullOrEmptyException(nameof(action));
-            if (string.IsNullOrWhiteSpace(controllerName)) throw new ArgumentNullOrEmptyException(nameof(controllerName));
+            if (action == null) throw new ArgumentNullException(nameof(action));
+            if (string.IsNullOrWhiteSpace(action)) throw new ArgumentException("Value can't be empty or consist only of white-space characters.", nameof(action));
+            if (controllerName == null) throw new ArgumentNullException(nameof(controllerName));
+            if (string.IsNullOrEmpty(controllerName)) throw new ArgumentException("Value can't be empty.", nameof(controllerName));
```

Comment:
This should be changed to `string.IsNullOrWhiteSpace()` 😇 

```suggestion
            if (string.IsNullOrWhiteSpace(controllerName)) throw new ArgumentException("Value can't be empty or consist only of white-space characters.", nameof(controllerName));
```

Reply:
indeed